### PR TITLE
Updates the notification area and introduces a toast mechanism

### LINKED ISSUE #840: Improve notifications
## Problem Definition

Currently, notifications are only displayed in the notification area which makes them really easy to miss. We should implement and add a mechanism that shows the notifications for a short period of time to the user. This will make it easier for the user to not miss important notification such as errors.

We need to add [popup notifications](https://en.wikipedia.org/wiki/Pop-up_notification).

We can use the https://vuetifyjs.com/en/components/snackbars/ for the popup notification.

By default, the snackbars are not stacking so we need to evaluate if it makes sense to implement a queue of the messages, or maybe create a summary and show one notification.

### PR DESCRIPTION
**Pull request by kelsos**
Closes #840

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -2,6 +2,7 @@
 Changelog
 =========
 
+* :feature: `840` Add a new notification UI. Backend errors should now display a notification on the upper right corner.
```
Comment (LefterisJP) [inline docs/changelog.rst:5]: Shouldn't you also specify difference between warnings and errors here?

File: frontend/app/src/components/status/notifications/Notification.vue [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  notification!: NotificationData;
+
+  @Emit()
+  dismiss(_notificationId: number) {}
+
+  icon(notification: NotificationData) {
+    switch (notification.severity) {
+      case Severity.ERROR:
+        return 'fa-exclamation-circle';
+      case Severity.INFO:
```
Comment (LefterisJP) [inline frontend/app/src/components/status/notifications/Notification.vue:86]: note: The backend has no info messages sent like that atm

File: frontend/app/src/components/status/notifications/NotificationPopup.vue [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+          />
+        </div>
+      </template>
+      <span>Number of Pending notifications</span>
+    </v-tooltip>
+  </v-snackbar>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Watch } from 'vue-property-decorator';
```
Comment (LefterisJP) [inline frontend/app/src/components/status/notifications/NotificationPopup.vue:35]: I don't know where but I guess somewhere in this file. Can you please make it that if the user clicks anywhere else the notifications sidebar goes away. It's really annoying to have to figure out that one has to press the top left `<` in order to have it move out of the way.

With notifications I would expect clicking anywhere else they would go away. And by clicking anywhere else I mean even at unclickable elements. Just any click somewhere else.

Peek 2020-08-18 13-31

File: frontend/app/src/components/status/notifications/Notification.vue (hunk 2 above)
Comment (kelsos) [reply frontend/app/src/components/status/notifications/Notification.vue:86]: There is an info message in the frontend so far.

File: frontend/app/src/components/status/notifications/NotificationPopup.vue (hunk 3 above)
Comment (kelsos) [reply frontend/app/src/components/status/notifications/NotificationPopup.vue:35]: Sure that should be relatively easy. It alredy works on the smaller screen sizes.

File: docs/changelog.rst (hunk 1 above)
Comment (kelsos) [reply docs/changelog.rst:5]: I am not sure it makes sense from a changelog perspective. Warnings didn't change that much from the previous version. There is just a slight visual change of the UI, but other than that they behave exactly the same way as before. Only the behavior for the backend errors changed.

Comment (LefterisJP) [reply docs/changelog.rst:5]: Okay np

**Pull request merged**